packages/apollo-server-integration-testsuite/src/index.ts

```diff
@@ -1206,22 +1361,47 @@ export default (createApp: CreateAppFunc, destroyApp?: DestroyAppFunc) => {
         expect(result.body.errors[0].extensions.code).toEqual(
           'PERSISTED_QUERY_NOT_FOUND',
         );
+
+        expect(didEncounterErrors).toBeCalledWith(
+          expect.objectContaining({
+            errors: expect.arrayContaining([
+              expect.any(PersistedQueryNotFoundError)
+            ]),
+          }),
+        );
       });
       it('returns result on the second try', async () => {
         await request(app)
           .post('/graphql')
           .send({
             extensions,
           });
+
+        // Only the first request should result in an error.
+        expect(didEncounterErrors).toBeCalledWith(
+          expect.objectContaining({
+            errors: expect.arrayContaining([
+              expect.any(PersistedQueryNotFoundError)
+            ]),
+          }),
+        );
+
         const result = await request(app)
           .post('/graphql')
           .send({
             extensions,
             query,
           });
 
+        // There should be no additional errors now.
+        expect(didEncounterErrors).toHaveBeenCalledTimes(1);
+
         expect(result.body.data).toEqual({ testString: 'it works' });
         expect(result.body.errors).toBeUndefined();
+
+        // Only the first request should have resulted in an error, not the
+        // second.
+        expect(didEncounterErrors).toHaveBeenCalledTimes(1);
```

Comment:
This expectation is duplicated immediately above

Reply:
Intentionally! Added comment to clarify: 21aa570a7b8a5aa9142033b94e36838dc03c240c.

Let me know if it makes sense?

Reply:
Hmm, in retrospect, you're right that this additional assertion may be confusing and unnecssary.  Maybe it would make more sense if I moved the first `expect`tation to be along-side the `expect(didEncounterErrors).toBeCalledWith(` on line 1381?

Reply:
See 66c32f8